spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWriteBuilder.java

```diff
@@ -81,6 +82,11 @@ public DeltaWrite build() {
         handleTimestampWithoutZone || !SparkUtil.hasTimestampWithoutZone(table.schema()),
         SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR);
 
+    if (!writeConf.branch().equalsIgnoreCase(SnapshotRef.MAIN_BRANCH)) {
+      throw new UnsupportedOperationException(
+          "Row level operations on non-main branches are currently unsupported");
+    }
```

Comment:
@rdblue @aokolnychyi @namrathamyske This will prevent any row level operations (temporarily until the SQL write config PR gets in) on non-main branches.

Reply:
I don't think we need a separate test for this. There is no way to set it anyway. If we want to have this validation, what about negating the sentence?

```
throw new UnsupportedOperationException(
    "Row-level operations are currently supported only on the main branch");
```

Reply:
+1 for Anton's suggestion.

Reply:
Yeah good point, it's usually clearer to have messages be in the "positive form"